sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1084,6 +1084,11 @@ class Analyzer(
           result
         case UnresolvedExtractValue(child, fieldExpr) if child.resolved =>
           ExtractValue(child, fieldExpr, resolver)
+        case f @ UnresolvedFunction(_, children, _, filter) if filter.isDefined =>
+          val newChildren = children.map(resolveExpressionTopDown(_, q))
+          val newFilter = filter.map{ expr => expr.mapChildren(resolveExpressionTopDown(_, q))}
```

Comment:
nit: format `.map{ ` => `.map { `

Reply:
Due to the above modifications, this problem does not exist.